File: app/src/main/java/com/jameschamberlain/footballteamtracker/StartupActivity.java
package com.jameschamberlain.footballteamtracker;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.jameschamberlain.footballteamtracker.fixtures.Fixture;

import java.util.ArrayList;
import java.util.Collections;

public class StartupActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        boolean isSetup = false;

        FileUtils.checkFiles(this);

        String teamName = FileUtils.readTeamFile();

        if (!teamName.isEmpty()) {
            isSetup = true;
        }

        if (isSetup) {
            ArrayList<Player> teamMembers = FileUtils.readPlayersFile();
            ArrayList<Fixture> fixtures = FileUtils.readFixturesFile();

            Team team = Team.getInstance();
            team.setName(teamName);
            team.setFixtures(fixtures);
            team.setPlayers(teamMembers);
            Intent newActivity = new Intent(this, MainActivity.class);
            this.startActivity(newActivity);
        } else {
            Intent newActivity = new Intent(this, SetupActivity.class);
            this.startActivity(newActivity);
        }

    }
}

File: app/src/main/java/com/jameschamberlain/footballteamtracker/StartupActivity.kt
package com.jameschamberlain.footballteamtracker

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.ktx.firestore
import com.google.firebase.ktx.Firebase
import com.jameschamberlain.footballteamtracker.objects.Team
import com.jameschamberlain.footballteamtracker.setup.SetupActivity

private const val TAG = "StartupActivity"

class StartupActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val isLoggedIn = FirebaseAuth.getInstance().currentUser != null

        if (isLoggedIn) {
            Log.i(TAG, "User logged in, progressing to main app")
            val preferences: SharedPreferences = getSharedPreferences("com.jameschamberlain.footballteamtracker", Context.MODE_PRIVATE)
            Utils.teamId = preferences.getString("team_id", null)!!
            Utils.teamRef = Firebase.firestore.document("teams/${Utils.teamId}")
            Utils.teamRef.get()
                    .addOnSuccessListener { documentSnapshot ->
                        Team.teamName = documentSnapshot.getString("name")!!
                        Utils.setupTeamListener()
                        startActivity(Intent(this, MainActivity::class.java))
                    }
                    .addOnFailureListener { e ->
                        Log.e(TAG, "Get failed with ", e)
                    }
        } else {
            startActivity(Intent(this, SetupActivity::class.java))
        }
    }
}